### PR TITLE
Feature/dwg read block attributes as feature fields

### PR DESCRIPTION
**Pull request by lchevallier69**
## What does this PR do?
Add new option in DWG Driver, DWG_ATTRIBUTES expose block attributes as feature attributes

## What are related issues/pull requests?
implementation of enhancement : #5035

### CODE REVIEW

**Comment by rouault** [comment]
It would be good if autotest/ogr/ogr_dwg.py could be enhanced to test this, with a sample file that is as small as possible (ideally a few 10's of KB maximum) and can be added to the test suite.

**Comment by lchevallier69** [comment]
> It would be good if autotest/ogr/ogr_dwg.py could be enhanced to test this, with a sample file that is as small as possible (ideally a few 10's of KB maximum) and can be added to the test suite.

I'll prepare this test

**Comment by lchevallier69** [comment]
> It would be good if autotest/ogr/ogr_dwg.py could be enhanced to test this, with a sample file that is as small as possible (ideally a few 10's of KB maximum) and can be added to the test suite.

I made a test that verify that the layerDefn contains new exposed attributes, check also some explicit attributes availabilty

File: autotest/ogr/ogr_dwg.py [hunk 1]
```diff
@@ -145,3 +145,27 @@ def test_ogr_dwg_3():
...
+        defn = layer.GetLayerDefn()
+        
+        assert defn.GetFieldCount() == 28, \
+            ('did not get expected number of fields in defn. got %d'
+                                % defn.GetFieldsCount())
+                                
+        assert defn.GetFieldIndex('AVAILABILITY') >-1, \
+            ('did not get the expected field in defn. AVAILABILITY')
+        assert defn.GetFieldIndex('ROOM') >-1, \
+            ('did not get the expected field in defn. ROOM')
```
Comment (rouault) [inline autotest/ogr/ogr_dwg.py:167]: perhaps also test the value of those attributes on a feature ?

feat = lyr.GetNextFeature()
assert feat['AVAILABILITY'] == something

Comment (lchevallier69) [reply autotest/ogr/ogr_dwg.py:167]: your're right, i'll make a new test

Comment (lchevallier69) [reply autotest/ogr/ogr_dwg.py:173]: done on 8cf4b87f21e0e74fe27cb6b62a669ca73da91c94

File: autotest/ogr/ogr_dwg.py [hunk 2]
```diff
@@ -145,3 +145,36 @@ def test_ogr_dwg_3():
...
+    with gdaltest.config_options({'DWG_INLINE_BLOCKS':'FALSE','DWG_ATTRIBUTES':'TRUE'}):
+
+        ds = gdal.OpenEx('data/cad/Building_A_Floor_0_Mapwize.dwg', allowed_drivers=['DWG'])
+
+        assert ds is not None
+
+        layer = ds.GetLayer( 'entities' )
+        defn = layer.GetLayerDefn()
+        
+        foundmathieu = None
```
Comment (rouault) [inline autotest/ogr/ogr_dwg.py:160]:
```suggestion
        foundMathieu = Fallse
```

File: autotest/ogr/ogr_dwg.py [hunk 3]
```diff
@@ -145,3 +145,36 @@ def test_ogr_dwg_3():
...
+        for feature in layer:
+            if feature.GetField("OCCUPANT") == 'Mathieu' :
+                foundmathieu = 'yes'
```
Comment (rouault) [inline autotest/ogr/ogr_dwg.py:163]:
```suggestion
                foundMathieu = True
```

File: autotest/ogr/ogr_dwg.py [hunk 4]
```diff
@@ -145,3 +145,36 @@ def test_ogr_dwg_3():
...
+        assert defn.GetFieldCount() == 28, \
+            ('did not get expected number of fields in defn. got %d'
+                                % defn.GetFieldsCount())
+                                
+        assert defn.GetFieldIndex('AVAILABILITY') >-1, \
+            ('did not get the expected field in defn. AVAILABILITY')
+        assert defn.GetFieldIndex('ROOM') >-1, \
+            ('did not get the expected field in defn. ROOM')
+
+        assert foundmathieu == 'yes', \
```
Comment (rouault) [inline autotest/ogr/ogr_dwg.py:175]:
```suggestion
        assert foundMathieu, \
```

**Review by rouault — COMMENTED** [review]
slightly more idiomatic using boolean type

File: ogr/ogrsf_frmts/dwg/ogr_dwg.h [hunk 5]
```diff
@@ -198,14 +201,17 @@ class OGRDWGDataSource final: public OGRDataSource
...
     void                AddStandardFields( OGRFeatureDefn *poDef );
 
     // Implemented in ogrdxf_blockmap.cpp
     void                ReadBlocksSection();
+    void                ReadAttDefinitions();
     OGRGeometry        *SimplifyBlockGeometry( OGRGeometryCollection * );
     DWGBlockDefinition *LookupBlock( const char *pszName );
     std::map<CPLString,DWGBlockDefinition> &GetBlockMap() { return oBlockMap; }
 
+    std::set<CPLString>& GetAttributes() { return attributeFields; };
```
Comment (rouault) [inline ogr/ogrsf_frmts/dwg/ogr_dwg.h:214]:
```suggestion
    std::set<CPLString>& GetAttributes() { return attributeFields; }
```

**Comment by lchevallier69** [comment]
Nothing more to add in this feature branch. @rouault  Many thanks for your review.

**Pull request merged**